marine_benchmark.c

```diff
@@ -0,0 +1,162 @@
+
+    size_t memory_start = get_current_rss();
+    clock_t start = clock();
+    for (int i = start_index; i < end_index; ++i) {
+        packet p = packets[i];
+        marine_result *packet_results = marine_dissect_packet(filter_id, (char *) p.data, p.header->len);
+        assert(packet_results->result == 1);
+        marine_free(packet_results);
+    }
+    clock_t end = clock();
+    size_t memory_end = get_current_rss();
+
+    float total_time = ((float) end - start) / CLOCKS_PER_SEC;
+    float pps = (float) PART_LEN / total_time;
+    float memory_usage = ((float) memory_end - memory_start) / 1024 / 1024;
+    printf("%d packets took: %f, which is its %f pps!\n The test took: %lf MB\n", PART_LEN, total_time, pps,
+           memory_usage);
+}
+
+int print_title(char *str) {
+    return printf("\n\033[4:1m%s\033[0m\n", str);
+}
+
+int main(int argc, char *argv[]) {
+    if (argc < 2) {
+        printf("You didn't pass a cap!\n");
+        printf("Have you run cap_maker.py to create the testing cap?\n");
+        printf("After generating the cap add the path as argument\n");
+        return -1;
+    }
+
+    char *cap_file = argv[1];
+    packet packets[PACKET_COUNT];
+    if (load_cap(cap_file, packets) < 0) {
+        printf("\nSomething went wrong\n");
+        return -1;
+    }
+
+    int part = 0;
+    char *bpf = "tcp port 4000 or tcp port 4001 or tcp port 4002 or tcp port 4003 or tcp port 4004 or tcp port 4005 or tcp port 4006 or tcp port 4007 or tcp port 4008 or tcp port 4009 or tcp port 4010 or tcp port 4011 or tcp port 4012 or tcp port 4013 or tcp port 4014 or tcp port 4015 or tcp port 4016 or tcp port 4017 or tcp port 4018 or tcp port 4019 or udp port 4000 or udp port 4001 or udp port 4002 or udp port 4003 or udp port 4004 or udp port 4005 or udp port 4006 or udp port 4007 or udp port 4008 or udp port 4009 or udp port 4010 or udp port 4011 or udp port 4012 or udp port 4013 or udp port 4014 or udp port 4015 or udp port 4016 or udp port 4017 or udp port 4018 or udp port 4019";
+    char *dfilter = "tcp.port == 4000 or tcp.port == 4001 or tcp.port == 4002 or tcp.port == 4003 or tcp.port == 4004 or tcp.port == 4005 or tcp.port == 4006 or tcp.port == 4007 or tcp.port == 4008 or tcp.port == 4009 or tcp.port == 4010 or tcp.port == 4011 or tcp.port == 4012 or tcp.port == 4013 or tcp.port == 4014 or tcp.port == 4015 or tcp.port == 4016 or tcp.port == 4017 or tcp.port == 4018 or tcp.port == 4019 or udp.port == 4000 or udp.port == 4001 or udp.port == 4002 or udp.port == 4003 or udp.port == 4004 or udp.port == 4005 or udp.port == 4006 or udp.port == 4007 or udp.port == 4008 or udp.port == 4009 or udp.port == 4010 or udp.port == 4011 or udp.port == 4012 or udp.port == 4013 or udp.port == 4014 or udp.port == 4015 or udp.port == 4016 or udp.port == 4017 or udp.port == 4018 or udp.port == 4019";
+    char *three_fields[] = {
+            "ip.proto",
+            "eth.dst",
+            "ip.host"
+    };
+    char *eight_fields[] = {
```

Comment:
Change out some fields for layer 4 fields (so we know for sure the dissector is called).